src/algebra/group/basic.lean

```diff
@@ -55,6 +55,30 @@ left_inv_eq_right_inv (trans (mul_comm _ _) hy) hz
 
 end comm_monoid
 
+section left_cancel_monoid
+
+variables {M : Type u} [left_cancel_monoid M]
+
+@[to_additive] lemma left_cancel_one {a : M} (h : a * a = a) : a = 1 :=
+mul_left_cancel (show a * a = a * 1, by rwa mul_one)
+
+@[to_additive] lemma one_left_cancel {a : M} (h : a = a * a) : a = 1 :=
+mul_left_cancel (show a * a = a * 1, by rwa [mul_one, eq_comm])
```

Comment:
Sure I'll change them but they are extremely long. I'd prefer then something like `left_cancel_to_one` or `left_cancel_eq_one`. Similar lemmas for groups are not in namespaces so I think it'd be better not to put these lemmas in namespaces since they'll mostly be used for groups and not for cancel monoids.

Reply:
I agree, short names are a good goal. Still, for searchability I would like `eq_one` and `mul_self` to appear in the names. And we need `left_cancel` to appear to avoid duplication with `left_cancel` and `right_cancel`. Maybe `eq_one_of_left_cancel_mul_self` and `eq_one_of_mul_self_left_cancel`?

Reply:
Ok